autogpt/commands/web_selenium.py

```diff
@@ -188,3 +199,34 @@ def add_header(driver: WebDriver) -> None:
         driver.execute_script(overlay_script)
     except Exception as e:
         print(f"Error executing overlay.js: {e}")
+
+
+def summarize_memorize_webpage(
+    url: str, text: str, question: str, driver: Optional[WebDriver] = None
+) -> str:
+    """Summarize text using the OpenAI API
+
+    Args:
+        url (str): The url of the text
+        text (str): The text to summarize
+        question (str): The question to ask the model
+        driver (WebDriver): The webdriver to use to scroll the page
+
+    Returns:
+        str: The summary of the text
+    """
+    if not text:
+        return "Error: No text to summarize"
+
+    text_length = len(text)
+    logger.info(f"Text length: {text_length} characters")
+
+    memory = get_memory(CFG)
+
+    if isinstance(memory, NoMemory):
```

Comment:
I think you should try to fix the interfaces so that this works without a special branch for `NoMemory` if it is not too much work.  I'm not sure why it doesn't work currently.  

At your discretion.

Reply:
It works, just costs a few millicents extra for creating embeddings that won't be used in the `NoMemory` backend. Nevertheless, I have removed the code because the savings are not that significant.